bundles/org.openhab.binding.folderwatcher/src/main/java/org/openhab/binding/folderwatcher/internal/handler/S3BucketWatcherHandler.java

```diff
@@ -0,0 +1,164 @@
+    private S3BucketWatcherConfiguration config = new S3BucketWatcherConfiguration();
+    private File currentS3ListingFile = new File(OpenHAB.getUserDataFolder() + File.separator + "FolderWatcher"
+            + File.separator + thing.getUID().getAsString().replace(':', '_') + ".data");
+    private @Nullable ScheduledFuture<?> executionJob;
+    private List<String> previousS3Listing = new ArrayList<>();
+    private @Nullable S3Client client;
+    private @Nullable AwsCredentialsProvider credentialsProvider;
+    private @Nullable Region region;
+
+    public S3BucketWatcherHandler(Thing thing) {
+        super(thing);
+    }
+
+    @Override
+    public void handleCommand(ChannelUID channelUID, Command command) {
+        logger.debug("Channel {} triggered with command {}", channelUID.getId(), command);
+        if (command instanceof RefreshType) {
+            refreshS3BucketInformation();
+        }
+    }
+
+    @Override
+    public void initialize() {
+        config = getConfigAs(S3BucketWatcherConfiguration.class);
+        updateStatus(ThingStatus.UNKNOWN);
+
+        try {
+            previousS3Listing = WatcherCommon.initStorage(currentS3ListingFile, config.S3BucketName);
+        } catch (IOException e) {
+            updateStatus(ThingStatus.OFFLINE, ThingStatusDetail.CONFIGURATION_ERROR, e.getMessage());
+            logger.debug("Can't write file {}: {}", currentS3ListingFile, e.getMessage());
+            return;
+        }
+        region = null;
+        for (Region reg : Region.regions()) {
+            if (reg.toString().equals(config.AWSRegion)) {
+                region = Region.of(config.AWSRegion);
+            }
+        }
+        if (region == null) {
+            updateStatus(ThingStatus.OFFLINE, ThingStatusDetail.CONFIGURATION_ERROR, "Please enter valid region");
+            return;
+        }
+        String accessKey = config.awskey;
+        String secretKey = config.awssecret;
+        AwsCredentials credentials;
+
+        if (config.Anonymous) {
+            credentialsProvider = AnonymousCredentialsProvider.create();
+        } else {
+            if (accessKey != null && !accessKey.isBlank() && secretKey != null && !secretKey.isBlank()) {
+                credentials = AwsBasicCredentials.create(accessKey, secretKey);
+                credentialsProvider = StaticCredentialsProvider.create(credentials);
+            }
+        }
+
+        if (refreshS3BucketInformation()) {
+            if (config.pollIntervalS3 > 0) {
+                updateStatus(ThingStatus.ONLINE);
+                executionJob = scheduler.scheduleWithFixedDelay(this::refreshS3BucketInformation, config.pollIntervalS3,
+                        config.pollIntervalS3, TimeUnit.SECONDS);
+            } else {
+                updateStatus(ThingStatus.OFFLINE, ThingStatusDetail.CONFIGURATION_ERROR,
+                        "Polling interval can't be null or negative");
```

Comment:
One might argue that 0 is negative (or not) :-), but i think the message should align to the check
'Polling interval must be greather then 0 seconds'